utils/build/docker/rust/parametric/src/main.rs

```diff
@@ -0,0 +1,260 @@
+use serde::Deserialize;
+use serde_json::json;
+use std::{
+    collections::HashMap,
+    env,
+    fmt::Display,
+    net::{IpAddr, Ipv4Addr},
+    panic,
+    sync::{Arc, Mutex, OnceLock},
+    time::Duration,
+};
+use time::{format_description::well_known::Rfc3339, OffsetDateTime};
+use tokio::{
+    net::TcpListener,
+    signal::unix::{signal, SignalKind},
+    time::sleep,
+};
+use tower::{Service, ServiceBuilder};
+use tower_http::trace::{DefaultOnFailure, DefaultOnRequest, DefaultOnResponse, TraceLayer};
+use tracing::{debug, error, field, info, info_span, Level, Span};
+use tracing_subscriber::{fmt, layer::SubscriberExt, util::SubscriberInitExt, EnvFilter};
+
+mod datadog;
+mod opentelemetry;
+
+pub(crate) fn get_tracer() -> &'static BoxedTracer {
+    static TRACER: OnceLock<BoxedTracer> = OnceLock::new();
+    TRACER.get_or_init(|| global::tracer("ddtrace-rust-client"))
+}
+
+#[derive(Clone)]
+struct AppState {
+    contexts: Arc<Mutex<HashMap<u64, Arc<ContextWithParent>>>>,
+    current_context: Arc<Mutex<Arc<ContextWithParent>>>,
+    extracted_span_contexts: Arc<Mutex<HashMap<u64, ::opentelemetry::Context>>>,
+    tracer_provider: Arc<SdkTracerProvider>,
+}
+
+#[derive(Default, Clone)]
+struct ContextWithParent {
+    context: ::opentelemetry::Context,
+    parent: Option<::opentelemetry::Context>,
+}
+
+impl ContextWithParent {
+    fn new(context: ::opentelemetry::Context, parent: Option<::opentelemetry::Context>) -> Self {
+        ContextWithParent { context, parent }
+    }
+}
+
+#[tokio::main]
+async fn main() {
+    // If tracing initialization fails, nevertheless emit a structured log event.
+    let result = init_tracing();
+    let tracer = match result {
+        Ok(tracer) => tracer,
+        Err(ref error) => {
+            log_error(error);
+            return;
+        }
+    };
+
+    // Replace the default panic hook with one that uses structured logging at ERROR level.
+    panic::set_hook(Box::new(|panic| error!(%panic, "process panicked")));
+
+    // Run and log any error.
+    if let Err(ref error) = run(Arc::new(tracer)).await {
+        error!(
+            error = format!("{error:#}"),
+            backtrace = %error.backtrace(),
+            "process exited with ERROR"
+        );
+    }
+}
+
+fn init_tracing() -> Result<SdkTracerProvider> {
+    let _ = tracing_subscriber::registry()
+        .with(EnvFilter::try_from_default_env().unwrap_or_else(|_| {
+            format!(
+                "{}=debug", // ,hyper=debug,tower_http=debug
+                env!("CARGO_CRATE_NAME")
+            )
+            .into()
+        }))
+        .with(fmt::layer().json().flatten_event(true))
+        .try_init()
+        .context("initialize tracing subscriber");
+
+    let mut builder = dd_trace::Config::builder();
+    builder.set_log_level_filter(dd_trace::log::LevelFilter::Debug);
+    Ok(datadog_opentelemetry::init_datadog(
+        builder.build(),
+        SdkTracerProvider::builder(),
+        None,
+    ))
```

Comment:
Could you replace the method of installation with this instead?
I'd like to remove `init_datadog` from the public API

```suggestion
    Ok(datadog_opentelemetry::tracing()
          .with_config(builder.build())
    )
```